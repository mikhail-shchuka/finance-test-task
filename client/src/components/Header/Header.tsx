import { useSelector } from 'react-redux';
import { RootState } from '../../redux/store';
import { TickerList } from '../TickerList';
import { FetchInterval } from '../FetchInterval';
import { Loader } from '../Loader';

import './Header.scss';

export const Header = () => {
  const tickersLoaded = useSelector((state: RootState) => state.tickersLoaded);

  return (
    <div className="header">
      <FetchInterval />
      {tickersLoaded ? <TickerList /> : <Loader />}
    </div>
  );
};
