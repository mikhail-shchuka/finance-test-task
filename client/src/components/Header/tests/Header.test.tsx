import { render, screen } from '@testing-library/react';
import { useSelector } from 'react-redux';
import { Header } from '../Header';

const mockFetchIntervalTestId = 'FetchInterval-test-id';
const mockTickerListTestId = 'TickerList-test-id';
const mockLoaderTestId = 'Loader-test-id';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

jest.mock('../../FetchInterval', () => {
  return {
    FetchInterval: () => (<div data-testid={mockFetchIntervalTestId} />),
  };
});

jest.mock('../../TickerList', () => {
  return {
    TickerList: () => (<div data-testid={mockTickerListTestId} />),
  };
});

jest.mock('../../Loader', () => {
  return {
    Loader: () => (<div data-testid={mockLoaderTestId} />),
  };
});

describe('Header', () => {
  beforeEach(() => {
    (useSelector as jest.Mock).mockClear();
  });

  it('should match with snapshot', () => {
    (useSelector as jest.Mock).mockImplementation(() => true);
    const { baseElement } = render(<Header />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should be rendered FetchInterval and TickerList if useSelector return true', () => {
    (useSelector as jest.Mock).mockImplementation(() => true);
    render(<Header />);

    expect(screen.getByTestId(mockFetchIntervalTestId)).toBeInTheDocument();
    expect(screen.getByTestId(mockTickerListTestId)).toBeInTheDocument();
  });

  it('should be rendered FetchInterval and Loader if useSelector return false', () => {
    (useSelector as jest.Mock).mockImplementation(() => false);
    render(<Header />);

    expect(screen.getByTestId(mockFetchIntervalTestId)).toBeInTheDocument();
    expect(screen.getByTestId(mockLoaderTestId)).toBeInTheDocument();
  });
});
