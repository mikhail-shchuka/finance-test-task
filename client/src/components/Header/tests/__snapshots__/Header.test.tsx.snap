// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match with snapshot 1`] = `
<body>
  <div>
    <div
      class="header"
    >
      <div
        data-testid="FetchInterval-test-id"
      />
      <div
        data-testid="TickerList-test-id"
      />
    </div>
  </div>
</body>
`;
